slsDetectorSoftware/src/Module.cpp

```diff
@@ -2609,41 +2536,16 @@ std::string Module::printReceiverConfiguration() {
 bool Module::getUseReceiverFlag() const { return shm()->useReceiverFlag; }
 
 int Module::lockReceiver(int lock) {
-    LOG(logDEBUG1) << "Setting receiver server lock to " << lock;
-    int retval = -1;
-    if (shm()->useReceiverFlag) {
-        sendToReceiver(F_LOCK_RECEIVER, lock, retval);
-        LOG(logDEBUG1) << "Receiver Lock: " << retval;
-    }
-    return retval;
+    return sendToReceiver<int>(F_LOCK_RECEIVER, lock);
 }
 
 sls::IpAddr Module::getReceiverLastClientIP() const {
-    sls::IpAddr retval;
-    LOG(logDEBUG1) << "Getting last client ip to receiver server";
-    if (shm()->useReceiverFlag) {
-        sendToReceiver(F_GET_LAST_RECEIVER_CLIENT_IP, nullptr, retval);
-        LOG(logDEBUG1) << "Last client IP from receiver: " << retval;
-    }
-    return retval;
+    return sendToReceiver<sls::IpAddr>(F_GET_LAST_RECEIVER_CLIENT_IP);
 }
 
 void Module::exitReceiver() {
     LOG(logDEBUG1) << "Sending exit command to receiver server";
-    if (shm()->useReceiverFlag) {
-        sendToReceiver(F_EXIT_RECEIVER, nullptr, nullptr);
-    }
-}
-
-void Module::execReceiverCommand(const std::string &cmd) {
-    char arg[MAX_STR_LENGTH]{};
-    char retval[MAX_STR_LENGTH]{};
-    sls::strcpy_safe(arg, cmd.c_str());
-    LOG(logDEBUG1) << "Sending command to receiver: " << arg;
-    if (shm()->useReceiverFlag) {
-        sendToReceiver(F_EXEC_RECEIVER_COMMAND, arg, retval);
-        LOG(logINFO) << "Receiver " << detId << " returned:\n" << retval;
-    }
+    sendToReceiver(F_EXIT_RECEIVER, nullptr, nullptr);
```

Comment:
can we not add sendToReceiver with only fnum, like you did for sendtodetector?

Reply:
Depends if we want to be able to do sendToReceiver<int>(fnum), we cant have both... since the only difference in signature is the return value overload resolution fails.

Reply:
ahh okay, then never mind, its too few in this case.